.github/workflows/auto-pr.yml

```diff
@@ -10,71 +10,130 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
+      - run: npm ci
 
       - name: Download extension
         env:
           BODY: ${{ github.event.issue.body }}
         run: |
           curl -L $(echo $BODY | grep -ioEh 'https?://\S+.zip' -m2 | tail -n1) -o /tmp/ext.zip
-          unzip -o /tmp/ext.zip -d ./extensions/community
 
-      - name: Rebuild the database
-        run: |
-          npm install
-          echo 'BUILD_LOGS<<EOF' >> $GITHUB_ENV
-          node scripts/generate-extensions-registry.js --disable-exit-code >> $GITHUB_ENV
-          echo 'EOF' >> $GITHUB_ENV
+      - name: Extract and verify extension
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { extractExtension } = require('./scripts/extract-extension.js');
+            const { verifyExtension } = require('./scripts/check-single-extension.js')
 
-      - name: Notify build errors
-        if: ${{ !contains(env.BUILD_LOGS, 'successfully updated') }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          gh issue comment ${{ github.event.issue.number }} --body "Hi @${{ github.actor }}!
 
-          Thanks for your submission! Unfortunately, our automated test system found a few issues with your extension:
+            const { error, extensionName } = await extractExtension();
```

Comment:
Pass the path to the zip file, otherwise from this point of view it looks like it's magically guessed (and could break without knowing for someone doing a change in this file).